packages/contracts/contracts/SmartInvoice.sol

```diff
@@ -130,6 +139,43 @@ contract SmartInvoice is
         emit Register(_client, _provider, amounts);
     }
 
+    function extendTerminationTime(uint256 _time) public {
+        require(!locked, "locked");
+        require(_msgSender() == client, "msg.sender !client");
+        terminationTime += _time;
+
+        emit TerminationExtension(_msgSender(), _time);
+    }
```

Comment:
I think extending the termination time is a completely separate feature that we didn't plan for. This would require discussion on our part as this changes a lot of assumptions about the system.